crates/core/src/parse.rs

```diff
@@ -688,7 +731,20 @@ fn build_addons(
             let repo_idx = repo_packages
                 .iter()
                 .position(|r| r.id == e.repository_id && r.kind == e.repository)?;
-            let repo_package = repo_packages.remove(repo_idx);
+
+            let mut repo_package = repo_packages.remove(repo_idx);
+
+            // Set the file id / version from the cache entry. This is needed to properly
+            // validate the installed version against the remote version
+            match &e.external_release_id {
+                Some(ExternalReleaseId::FileId(file_id)) => {
+                    repo_package.metadata.file_id = Some(*file_id)
+                }
+                Some(ExternalReleaseId::Version(version)) => {
+                    repo_package.metadata.version = Some(version.clone())
+                }
+                None => {}
+            }
```

Comment:
Here is where we apply that cached remote release version when building addon from cache entry